src/hooks/useTransactionSteps.ts

```diff
@@ -59,27 +64,23 @@ const useTransactionSteps = ({ transaction, onComplete }: Props) => {
     bestBlockFinalized: bestBlockFinalizedOnTarget
   } = useDashboard(targetRole);
 
-  useEffect(() => {
-    const getNonceOfTargetFinalizedBlock = async () => {
-      const nonce = await getNonceByHash(parseInt(targetMessageDeliveryBlock));
-      setTransactionNonceOfTargetFinalizedBlock(nonce);
-    };
-
-    if (targetMessageDeliveryBlock) {
-      getNonceOfTargetFinalizedBlock();
-    }
-  }, [targetMessageDeliveryBlock, getNonceByHash, setTransactionNonceOfTargetFinalizedBlock]);
-
   useEffect(() => {
     if (!areApiLoading || !transaction || finished) {
       return;
     }
 
-    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null): boolean => {
+    const stepEvaluator = (
+      transactionValue: string | number | null,
+      chainValue: string | number | null,
+      greaterEqual?: boolean
```

Comment:
This should be the only option. I think in all cases it's enough when we have `>=`.